fix(core): Re-assign error codes to be within core bounds (<1000)

danieljancar opened the pull request:
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/main/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)


## PR Type
What kind of change does this PR introduce?

<!-- Please check the one that applies to this PR using "x". -->

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:


## What is the current behavior?
<!-- Please describe the current behavior that you are modifying, or link to a relevant issue. -->

Two error codes in the angular/core package incorrectly use error codes 1000 and 1001, which should be used for the reserved range in the angular/forms package.

```txt
Error code ranges per package:
 - core (this package): 100-999
 - forms: 1000-1999
 - common: 2000-2999
 - animations: 3000-3999
 - router: 4000-4999
 - platform-browser: 5000-5500
 ```


Issue Number: #53433 


## What is the new behavior?

This PR reassigns the error codes for RUNTIME_DEPS_INVALID_IMPORTED_TYPE and RUNTIME_DEPS_ORPHAN_COMPONENT from 1000 and 1001 to 950 and 951, to align with the reserved range for the core package.


## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No


<!-- If this PR contains a breaking change, please describe the impact and migration path for existing applications below. -->


## Other information

Close #53433

JeanMeche commented on packages/core/src/errors.ts:
```diff
@@ -118,8 +118,8 @@ export const enum RuntimeErrorCode {
-  RUNTIME_DEPS_INVALID_IMPORTED_TYPE = 1000,
-  RUNTIME_DEPS_ORPHAN_COMPONENT = 1001,
+  RUNTIME_DEPS_INVALID_IMPORTED_TYPE = 950,
+  RUNTIME_DEPS_ORPHAN_COMPONENT = 951,
```
Nit: 
Can we maybe add a comment that 999 is the upper limit for this file.
You know, to prevent someone to make that error again !

danieljancar replied:
Yeah, good input!

JeanMeche commented:
Hi, thanks for helping us with that change ! 

We also need to update our api golden files. Can you run `yarn bazel run //packages/core:core_errors.accept` ?

JeanMeche commented:
Could you please squash your commits into one and make sure the remaining one has a `refactor(core): ....` commit message. 

Thx for the help !

danieljancar commented:
Hey @JeanMeche,
I squashed the commits, should be ready now. 🚀

JeanMeche commented:
Hey, can you drop that merge commit ? We'd like the change to be a single commit !

danieljancar commented:
Hey, yeah is squashed. 😉

JeanMeche commented:
It looks like there is a conflict now, care to have a look at it ?

danieljancar commented:
@JeanMeche somehow I closed the PR when squashing, now had to reopen to fix the merge conflicts. 🦖

danieljancar commented:
I noticed a merge conflict from main branch where new error codes in the close range, but with `-` operators, have been added. I fixed merge conflict and just kept it that way:

```ts
<<<<<<< 53433-fix-error-code-beyond-expected-range
    RUNTIME_DEPS_INVALID_IMPORTED_TYPE = 950,
=======
    REQUIRED_MODEL_NO_VALUE = -952,
    // (undocumented)
    REQUIRED_QUERY_NO_VALUE = -951,
    // (undocumented)
    RUNTIME_DEPS_INVALID_IMPORTED_TYPE = 1000,
>>>>>>> main
```

JeanMeche commented:
Error codes with a minus denote an error that has an error page. For example : https://angular.io/errors/NG0950
So you're  going to need to increment the error code. 

Once this is done, you will need to update the error goldens with `yarn bazel run //packages/core:core_errors.accept`. 
Also keep in mind that this change should be a single commit and we will need to drop that merge commit.

danieljancar committed:
fix(core): Re-assign error codes to be within core bounds (<1000)

`RUNTIME_DEPS_INVALID_IMPORTED_TYPE` is now 980
`RUNTIME_DEPS_ORPHAN_COMPONENT` is now 981

danieljancar commented:
Hey Jean, thanks for your fast response. I increased the `RUNTIME_DEPS_...` to start at `960` and will update/ready my PR as soon as I'm on a better network since Bazel throws `timed-out` errors because my school network is so restrictive. 🥲

danieljancar commented:
@JeanMeche ran the goldens, should finally be ready. 🚀

danieljancar commented:
No idea whats going on with the local formatting recently, either the previous configs from other contributors didn't go through correctly and now I'm formatting with the real configs, since I set manual configuration on the `node-modules` prettier package in my IDE. 🥲

JeanMeche commented:
The angular code base hasn't fully migrated to prettier. The core package still uses clang to format.

danieljancar commented:
Yeah, my IDE formatting settings, ng-dev and the other config files don't go well together. Fixed the problem for now.

JeanMeche commented:
That last failing test is a flake. Change looks good now !

AndrewKushnir commented:
@danieljancar thanks for creating this PR! It looks like one of the error codes was used inside of some tests, so they are failing on CI, [see here](https://github.com/angular/angular/actions/runs/8066490051/job/22035217456?pr=53455). Could you please take a look and make the necessary updates when you get a chance? Thank you.

JeanMeche commented:
Hi, it looks like this needs a rebase again.

alxhub commented:
Hi @danieljancar,

can you adjust your commit message to address the lint failure, and rebase this PR?

JeanMeche commented:
OP seemed a bit busy and since I already had to mention this issue into other PRs, I took the matter.

danieljancar commented:
Thanks @JeanMeche 🙏🏻

AndrewKushnir reviewed (approved):
Reviewed-for: public-api

AndrewKushnir reviewed (approved):
Reviewed-for: public-api

thePunderWoman reviewed (approved):
reviewed-for: public-api

AndrewKushnir commented:
This PR was merged into the repository by commit 656b5d3e78004229a76488e0de1eb1d3508d8f6d.

The changes were merged into the following branches: main
